### PR TITLE
Added some autoloads

### PR DESCRIPTION
**Pull request by vmiheer**
I see autoloads sprinkled in many core modes like c++-mode or c-mode. And I think it increases usability of the mode.

### CODE REVIEW

File: promela-mode.el [hunk 1]
```diff
@@ -437,8 +437,9 @@ If (match-beginning 2) is non-nil, the item is followed by a `value'."
 
 
 ;; -------------------------------------------------------------------------
+;;;###autoload (add-to-list 'auto-mode-alist '("\\.\\(promela\\|spin\\|pml\\)\\'" . promela-mode))
```
Comment (rudi) [inline promela-mode.el:440]: I think there should be a line break between the autoload cookie and the `add-to-list` form

Comment (vmiheer) [reply promela-mode.el:440]: I got that inspiration from: https://github.com/emacs-mirror/emacs/blob/master/lisp/progmodes/cc-mode.el#L2657
Also I have tested the changes locally using doom emacs (straight.el).

**Comment by rudi** [comment]
See the Emacs Lisp manual, node 16.5 ("Autoload"):

>    A magic autoload comment (often called an “autoload cookie”) consists of `;;;###autoload`, _on a line by itself_, just before the real definition of the function in its autoloadable source file.

(emphasis added)

Omitting the line break means the (add-to-list ...) form is inside a comment and will not be executed.  Maybe the `update-file-autoload` command still handles this case, but loading the file does not.  Have you tried to load the file and verified that the entry for promela-mode was added to `auto-mode-alist`?

(Otherwise I think your changes are good; happy to merge the pull request once this is fixed.)

**Comment by vmiheer** [comment]
Okay here's what I did:
1. start emacs: `emacs -Q`.
2. `(load-file "c:/Users/vmihe/source/repos/promela-mode/promela-mode.el")` in `*scratch*` buffer.
3. `describe-variable` (`C-h v`) auto-mode-alist

It's:
```lisp
(("\\.pml\\'" . promela-mode)
 ("\\.spin\\'" . promela-mode)
 ("\\.promela\\'" . promela-mode)
...
```

Is this the same thing when you meant:
> Have you tried to load the file and verified that the entry for promela-mode was added to auto-mode-alist?

I am using emacs 27.2.

**Comment by vmiheer** [comment]
Gentle reminder if my earlier comment answers to your question.

**Comment by rudi** [comment]
Sorry, I was traveling over the weekend, will get to it today.

**Comment by vmiheer** [comment]
No worries at all :).

**Comment by rudi** [comment]
Hi, I believe the additions to the `auto-mode-alist` variable (after loading promela-mode.el) are done at Lines 545-547 :)  When I comment out these lines and `M-x eval-buffer` the file, `auto-mode-alist` doesn't contain entries for promela-mode.  Will fix.

**Comment by rudi** [comment]
Note that when you loaded the file, auto-mode-alist contained three new entries and not the one more expressive entry your patch aimed to add.  I stayed with the three separate entries since that's what other modes seem to use in similar situations, e.g., image-mode.  Thanks again for your contribution!  The mode is better now for sure.

**Pull request merged**